### PR TITLE
Embedded groups

### LINKED ISSUE #1783: Embedding group pages
Users want to be able to embed group pages into their own websites. This feature should be based on the existing work around [embedding shopfronts](https://github.com/openfoodfoundation/openfoodnetwork/pull/1557), with some modifications.

@mkllnk I understand the plan is that you should spend 2 hours on this, and see how far we can get. I've listed some 'nice to have' elements, but it's ok if they don't get done in the first version.

**Group page functionality to keep the same:** (essential)
- Existing map search and zoom
- Existing search and filters on the profiles and hubs tabs
- Expando collapso of the mini profiles in Our Producers and Our Hubs

**Display of profile modals** (not essential)
When a visitor clicks on a profile icon on the map, the modal should open. Ideally the background of the webpage (not just the embedded section) should be greyed (as pictured below). The webpage scroll bar is also disabled, so the user can just scroll in the modal.
image
If this is not possible, just the embedded section should be greyed, as happens if you open the product modal in an embedded shop - see [here ](https://openfoodnetwork.org/user-guide/advanced-features/demo-embedded-shop/)

**Embedded header** (not essential)
- I'd suggest that the Login (or gear symbol if already logged in) and Cart links could be removed from the header- I imagine the directory will primarily be for the purpose of browsing profiles, not shopping. Therefore we don't need these functions to be at the top, where they'll potentially confuse people.
- Could we move the 'powered by open food network' to somewhere lower on the page? It takes up a whole row of header space, which is probably not neccessary.
image

**Contact details** (not essential)
If possible it would make sense to remove the contact details 
image

**Remove the logo, banner and Description text fields.** (not essential)
Because the group page is being embedded, the group owner can add whatever images and text they want above their embedded group. Therefore if we remove these fields, it gives the group owner more flexibility to decide what sits above the directory.
image

**Links to Shops.** (not essential)
If a profile modal for a shop contains a link to their OFN shop, it would be great if this could open in a new tab, rather than regular functionality, which directs the user to the shop in their current tab. These links are found on the 1) map 2) Our Producers tab and 3) Our Hubs tab.
image

### PR DESCRIPTION
**Pull request by KeirOsborn**
Closes #1783 

Six commits. Two for initial tests and setup and four for each of the remaining sub-issues within this issue. The sub issues in question are:

1. move the 'powered by open food network' to somewhere lower on the page (the element got removed)
- It just adds a small bit of text to the group template saying "Powered by open food network". This only appears when a group page is embedded.

2. remove the contact details.
- It removes the contact details from a group page when that page is being viewed in an embedded setting.

3. Remove the logo, banner and Description text fields.
- It removes the whole header element from from the group page template if it's being viewed as an embedded page. This also includes the group name so that has also been removed (as instructed).

4. Links to shops open a new window.
- If a group page is being viewed as an embedded page, all links to the groups corresponding shops are served with the target = '_blank' attribute so that they open in a new browser window/tab.

All of these are solved by this PR. 

#### What should we test?

Tests in spec/features/consumer/shopping/embedded_groups_spec.rb

Also functionality can be manually tested using the embedded-group-preview.html file. i.e. by creating a group and then navigating to: http://localhost:3000/embedded-group-preview.html?your-group-name.

_Screenshots for comparison:_

Standard show view:
screen shot 2018-02-01 at 12 48 20

Embedded view:
screen shot 2018-02-01 at 12 48 00

#### Discourse thread

Discussion here:
https://github.com/openfoodfoundation/openfoodnetwork/issues/1783
and here:
https://github.com/openfoodfoundation/openfoodnetwork/pull/2031

### CODE REVIEW

**Comment by enricostano** [comment]
Cool, thanks @KeirOsborn 👏 

Two questions:
 - do you mind 2 screenshots of the `embedded` and `show` views added to the description?
 - is there anything we can re-use via partials between `embedded` and `show` views?

File: app/assets/stylesheets/darkswarm/embedded_shopfront.css.scss [hunk 1]
```diff
@@ -51,6 +51,16 @@ nav.top-bar ul.right li.powered-by {
   }
 }
 
+.powered-by-embedded {
+   opacity: 0.6;
+   font-family: "Oswald", sans-serif;
```
Comment (sauloperez) [inline app/assets/stylesheets/darkswarm/embedded_shopfront.css.scss]: don't we inherit this already? If I'm not mistaken we use this typeface in other places. If that wasn't the case I'd go with a sass variable. thoughts?

File: spec/features/consumer/shopping/embedded_groups_spec.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
+require 'spec_helper'
+
+feature "Using embedded shopfront functionality", js: true do
+
+  Capybara.server_port = 9999
```
Comment (sauloperez) [inline spec/features/consumer/shopping/embedded_groups_spec.rb]: Can you check if this is already configured in `spec/spec_helper.rb`? To me, it's too unrelated to this test file to have it here and it looks like a global `Capybara` config.

**Comment by KeirOsborn** [comment]
@enricostano 

- do you mind adding 2 screenshots of the embedded and show views to the description?

Sure, have added. 

- Is there anything we can re-use via partials between embedded and show views?

I don't think so. The problem is that the embedded version needs to have target="_blank" each time there is a link back to the shop. These are scattered throughout the partials so most of them have to be re-written. Unless anyone knows a way to set target="_blank" for a specific link at the controller level. I don't.

**Comment by enricostano** [comment]
@KeirOsborn 

> Unless anyone knows a way to set target="_blank" for a specific link at the controller level

IF that's (almost) the only difference maybe we can pass a `local` variable to the view like `target` and set it in the controller:

`target: @target` instead of `target: "_blank"`

It can also be a view helper probably. Is that the main difference?

File: app/assets/stylesheets/darkswarm/embedded_shopfront.css.scss (hunk 1 above)
Comment (KeirOsborn) [reply app/assets/stylesheets/darkswarm/embedded_shopfront.css.scss]: @sauloperez 

Cool. Agreed. This does exist as a sass variable already so i've used that instead. Have updated in latest version which I will push shortly.

**Comment by KeirOsborn** [comment]
@enricostano 

Ah yes. Thanks for the advice. Not sure why I didn't think of that. Ok I've done that now. See latest version of the pull request: #2066.

Thanks

**Pull request closed without merging**